libs/velo-external-db-core/src/web/jwt-verifier-decoder-middleware.ts

```diff
@@ -0,0 +1,22 @@
+import { Request, Response, NextFunction } from 'express'
+import { JWTVerifier } from './jwt-verifier'
+
+export class JWTVerifierDecoderMiddleware {
+    private jwtVerifier: JWTVerifier
+
+    constructor(jwtVerifier: any) {
+        this.jwtVerifier = jwtVerifier
+    }
+
+    verifyAndDecodeMiddleware() {
+        return (req: Request, res: Response, next: NextFunction) => {
+            try {
+                req.body = this.jwtVerifier.verifyAndDecode(req.body)
+                next()
+            }
+            catch (error) {
+                next(error)
+            }
+        }
+    }
+}
```

Comment:
I don't think you need a function that returns a function here, if we will use this function it as middleware we can just write `<instanceNameOfJWTVerifierDecoderMiddleware>.verifyAndDecodeMiddleware`